packages/aws-cdk/lib/cli-arguments.ts

```diff
@@ -0,0 +1,1253 @@
+// -------------------------------------------------------------------------------------------
+// GENERATED FROM packages/aws-cdk/lib/config.ts.
+// Do not edit by hand; all changes will be overwritten at build time from the config file.
+// -------------------------------------------------------------------------------------------
+/* eslint-disable @typescript-eslint/comma-dangle, comma-spacing, max-len, quotes, quote-props */
+/**
+ * The structure of the CLI configuration, generated from packages/aws-cdk/lib/config.ts
+ *
+ * @struct
+ */
+export interface CliArguments {
+  /**
+   * The CLI command name followed by any properties of the command
+   */
+  readonly _: Array<string>;
```

Comment:
This is supposed to be typed `[Command, ...string]` but I don't have a good way of doing that via code-gen (enlighten me if you think differently). Instead I plan to validate that the first array element is a valid Command when I am converting the input into this type.

Reply:
we might also want to pull out the additional elements into its own `STACKS` property, as that is what the current `Arguments` type does:

```ts
export type Arguments = {
  readonly _: [Command, ...string[]];
  readonly exclusively?: boolean;
  readonly STACKS?: string[];
  readonly lookups?: boolean;
  readonly [name: string]: unknown;
};
```

Reply:
1) You could just not include it hear, keep this interface to just the auto-generated stuff and inside the CLI have an interface extending the codegen'd one with whatever extra types you need.

2) Misusing `ambient` should work: `Type.ambient('[Command, ...string[]]')` You'll have to import `Command` manually though.